server/block/lectern.go

```diff
@@ -0,0 +1,163 @@
+package block
+
+import (
+	"fmt"
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"time"
+)
+
+// Lectern is a librarian's job site block found in villages. It is used to hold books for multiple players to read in
+// multiplayer.
+// TODO: Redstone functionality.
+type Lectern struct {
+	bass
+	sourceWaterDisplacer
+
+	// Facing represents the direction the Lectern is facing.
+	Facing cube.Direction
+	// Book is the book currently held by the Lectern.
+	Book item.Stack
+	// Page is the page the Lectern is currently on in the book.
+	Page int
+}
+
+// Model ...
+func (Lectern) Model() world.BlockModel {
+	return model.Lectern{}
+}
+
+// FuelInfo ...
+func (Lectern) FuelInfo() item.FuelInfo {
+	return newFuelInfo(time.Second * 15)
+}
+
+// SideClosed ...
+func (Lectern) SideClosed(cube.Pos, cube.Pos, *world.World) bool {
+	return false
+}
+
+// BreakInfo ...
+func (l Lectern) BreakInfo() BreakInfo {
+	d := []item.Stack{item.NewStack(Lectern{}, 1)}
+	if !l.Book.Empty() {
+		d = append(d, l.Book)
+	}
+	return newBreakInfo(2, alwaysHarvestable, axeEffective, simpleDrops(d...))
+}
+
+// UseOnBlock ...
+func (l Lectern) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.World, user item.User, ctx *item.UseContext) (used bool) {
+	pos, _, used = firstReplaceable(w, pos, face, l)
+	if !used {
+		return false
+	}
+	l.Facing = user.Facing().Opposite()
+	place(w, pos, l, user, ctx)
+	return placed(ctx)
+}
+
+// readableBook represents a book that can be read through a lectern.
+type readableBook interface {
+	// TotalPages returns the total number of pages in the book.
+	TotalPages() int
+	// Page returns a specific page from the book and true when the page exists. It will otherwise return an empty string
+	// and false.
+	Page(page int) (string, bool)
+}
+
+// Activate ...
+func (l Lectern) Activate(pos cube.Pos, _ cube.Face, w *world.World, u item.User, ctx *item.UseContext) bool {
+	if !l.Book.Empty() {
+		// We can't put a book on the lectern if it's full.
+		return true
+	}
+	held, _ := u.HeldItems()
+	if _, ok := held.Item().(readableBook); ok {
+		l.Page = 0
+		l.Book = held
+
+		w.SetBlock(pos, l, nil)
+		w.PlaySound(pos.Vec3Centre(), sound.LecternBookPlace{})
+		ctx.SubtractFromCount(1)
+	}
+	return true
```

Comment:
You should probably return false if item is not a book or if the Lectern already held a book (first check).